rhel/vex/fetcher.go

```diff
@@ -0,0 +1,288 @@
+package vex
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/klauspost/compress/snappy"
+	"github.com/klauspost/compress/zstd"
+	"github.com/quay/zlog"
+
+	"github.com/quay/claircore/libvuln/driver"
+	"github.com/quay/claircore/pkg/tmp"
+)
+
+// A Changes is a series of Entries indicating where to find resources in the
+// changes.csv file.
+type Changes []Entry
+
+// Entry is an entry in a changes.csv file.
+type Entry struct {
+	// This path should be parsed in the context of the manifest's URL.
+	Path        string
+	updatedTime time.Time
+}
+
+func (u *VEXUpdater) Fetch(ctx context.Context, hint driver.Fingerprint) (io.ReadCloser, driver.Fingerprint, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "rhel/VEXUpdater.Fetch")
+	fp, err := ParseFingerprint(string(hint))
+	if err != nil {
+		return nil, hint, err
+	}
+
+	f, err := tmp.NewFile("", "rhel-vex.")
+	if err != nil {
+		return nil, hint, err
+	}
+
+	cw := snappy.NewBufferedWriter(f)
```

Comment:
just curious what made you choose snappy?

Reply:
It satisfied the need for:
> very high speeds and reasonable compression
i.e. the actual compression doesn't save us much apart from the spool being smaller and it being structured JSON there is most likely a lot of low hanging fruit anyway for compression so spending extra effort to compress it the smallest possible didn't seem worth the trade-off.